### PR TITLE
--no-container no longer ignored by cwltoil

### PR DESCRIPTION
**Pull request by guillermo-carrasco**
There has been a discussion in Common Workflow Language [Gitter](https://gitter.im/common-workflow-language/common-workflow-language) channel about `cwltoil` not passing around the `--no-container` option. This causes that even if you explicitely specify `--no-container`, if there is a `DockerRequirement` hint in your workflow, `cwltool` will try to use Docker. This PR should solve the problem.

As an addition, thanks to the use of `kwargs`, its possible to extend the options passed to `cwltool` if desired. 

@tetron could you please give this a look? Its tested in our `HPC` and working (the option, not the entire workflow yet).

Thanks!

### CODE REVIEW

File: src/toil/cwl/cwltoil.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def run(self, fileStore):
 class CWLJob(Job):
     """Execute a CWL tool wrapper."""
 
-    def __init__(self, tool, cwljob):
+    def __init__(self, tool, cwljob, **kwargs):
```
Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: Why not just `use_container=True` instead?

Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: Scratch that.

File: src/toil/cwl/cwltoil.py [hunk 2]
```diff
@@ -192,6 +192,7 @@ def __init__(self, tool, cwljob):
         #super(CWLJob, self).__init__()
         self.cwltool = tool
         self.cwljob = cwljob
+        self.use_container = kwargs.get('use_container', True)
```
Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: `self.kwargs  = kwargs`

File: src/toil/cwl/cwltoil.py [hunk 3]
```diff
@@ -213,31 +214,32 @@ def run(self, fileStore):
                                                   os.getcwd(), None,
                                                   outdir=outdir,
                                                   tmpdir=tmpdir,
-                                                  use_container=True)
+                                                  use_container=self.use_container)
```
Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: `**self.kwargs`

Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: And maybe rename `kwargs` and `self.kwargs` to `(self.)executor_options or (self.)run_options`

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by guillermo-carrasco** [comment]
Thanks for the good suggestions @hannes-ucsc , my last commit should amend that, I've tested it in our cluster.

**Comment by guillermo-carrasco** [comment]
@hannes-ucsc or @tetron , could you please have a look at this and tell me if there is something else I can do?

Thanks a lot!

File: src/toil/cwl/cwltoil.py (hunk 1 above)
Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: You probably want to name this argument `executor_options`, too. Or document this argument in the docstring.

File: src/toil/cwl/cwltoil.py [hunk 4]
```diff
@@ -192,6 +192,7 @@ def __init__(self, tool, cwljob):
         #super(CWLJob, self).__init__()
         self.cwltool = tool
         self.cwljob = cwljob
+        self.executor_options = kwargs
```
Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: For backwards compatibility, you would need to ensure that `use_container` defaults to True if it is absent from the dictionary. But wait a minute. Actually, I would only default use_container if kwargs is empty:

```
self.executor_options = kwargs or dict( use_container, True )
```

But I don't know if you want to be backwards compatible. @tetron?

Comment (hannes-ucsc) [inline src/toil/cwl/cwltoil.py]: Sorry, scratch all that. I read further down in your PR and it is better the way it is.

**Comment by hannes-ucsc** [comment]
+1 from me, except maybe renaming kwargs to executor_options.

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by hannes-ucsc** [comment]
Waiting for @tetron's approval as the maintainer of the cwltoil integration.

**Comment by guillermo-carrasco** [comment]
Great, thanks @hannes-ucsc :-)

**Comment by benedictpaten** [comment]
@tetron Are we happy to merge this?

**Comment by tetron** [comment]
@guillermo-carrasco @hannes-ucsc @benedictpaten yes looks good to me

**Comment by guillermo-carrasco** [comment]
Fantastic, thanks!

**Pull request merged**